### ISSUE TITLE
Not able to set hostname to the sitemaps

### ISSUE THREAD

**Issue by kishori-patil1** [nuxt-modules/sitemap#262]
`@nuxtjs/sitemap`: `5.0.5`
`nuxt`: `3.9.1`

The sitemap configuration:

> sitemap: {
    sitemaps: true,
    autoI18n: false,
    sources: ["/api/sitemap"],
  },

This is producing the sitemaps but the hostname is different.
<img width="981" alt="Screenshot 2024-04-09 at 8 23 08 PM" src="https://github.com/nuxt-modules/sitemap/assets/59562917/99d6d5b0-8b6a-4a8e-b4fc-f85a7e51bea1">

In the screenshot that I have attached, if you can see the hostname of app is different and the sitemap which are generated with the internal URLs. And the weird thing is, the internal url is repeated twice in the sitemap urls. I don't know why this is happening.

I am wondering if there is any way to add the hostname to the sitemaps.

**Comment by Bakusedu** [comment]
@kishori-patil1 try adding the following to your `nuxt.config.ts` to add your hostname:
```javascript
site: {
	url: process.env.BASE_URL, // url to your nuxt app
},
``` 
See: https://nuxtseo.com/sitemap/getting-started/installation

<img width="818" alt="Screenshot 2024-04-09 at 16 16 45" src="https://github.com/nuxt-modules/sitemap/assets/34550806/6efa87e2-c1cc-4a20-bb2f-2c13bd1b5d60">

**Comment by kishori-patil1** [comment]
@Bakusedu Hi, I tried this. But still I am getting the similar output.
Well, I am setting the host when I run the application. so it is taking only that value.
Is there any other way where we can add the different/multi domain urls to the sitemap?

**Comment by kishori-patil1** [comment]
@Bakusedu @harlan-zw Hi, One thing I have observed that, the `sources` array in sitemap config object, it fetches the sitemaps from the external endpoints which are mentioned in the array.
So when I visit the `/sitemap_index.xml`, The request call gets hit. Now, I have all the data, but when I want to see, let's say `0-sitemap.xml`, the request call again gets hit. In my understanding, this should not happen since you already have the data.
Can you please help me understanding the behaviour of `sources` config as well as can we add `sources` for each sitemap in Automatic chunking?

**Comment by chenyuncai** [comment]
the config field "site.url " in nuxt.config is always prefixed in final sitemap links, but maybe it is not needed for all case.

On my case, we deploy one application with many  sub-domains.

solution1: remove site.url in nuxt.config.ts and return loc with full href in dynamic endpoint
solution2: need sitemap module to avoid to add site.url again when links already has "http" prefix.

**Comment by chenyuncai** [comment]
can we pass the original H3Event to dynamic source fetch in  buildSitemap -> resolveSitemapSources -> fetchDataSource and proxy more headers like "host"

**Comment by harlan-zw** [comment]
I have tried to replicate both issues here:
- Absolute URLs getting prefixed with the site URL
- Absolute URLs hostname getting stripped for the site path

I was not able to replicate either, please provide a Stackblitz reproduction if you'd like me to explore further. See https://nuxtseo.com/og-image/getting-started/stackblitz

If you have feature requests outside of these issues please make them as their own issue so they can be tracked and discussed properly.

**Comment by chenyuncai** [comment]
I reproduce and check the link below.

1. The log in dynamic endpoint can only get 'localhost',
2. When i18n enabled, the returned urls are prefixed with site.url

https://stackblitz.com/edit/nuxt-starter-w4gkcw?file=server%2Fapi%2Fsitemap.ts

**Comment by harlan-zw** [comment]
Thank you @chenyuncai, seems like this was a specific issue with absolute URLs in dynamic sources with i18n enabled.

Should be fixed in [v5.1.5](https://github.com/nuxt-modules/sitemap/releases/tag/v5.1.5).
